### PR TITLE
Fix learn error calculation for logloss with monotonic constraints.

### PR DESCRIPTION
**Pull request by Andrew-Angrew**
Fix learn approx update when model shrinkage is on and approxes are exponentiated.

### CODE REVIEW

File: catboost/private/libs/algo/train.cpp [hunk 1]
```diff
@@ -77,8 +77,8 @@ static void ScaleAllApproxes(
         }
     }
     allApproxes.push_back(&learnProgress->AveragingFold.BodyTailArr[0].Approx);
-    allApproxes.push_back(&learnProgress->AvrgApprox);
     const int learnApproxesCount = SafeIntegerCast<int>(allApproxes.size());
+    allApproxes.push_back(&learnProgress->AvrgApprox);
```
Comment (Andrew-Angrew) [inline catboost/private/libs/algo/train.cpp]: Теперь эти аппроксы обновятся как не "экспоненцированные".

**Comment by annaveronika** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
CLA already signed

**Comment by arcadia-devtools** [comment]
@annaveronika, internal review request created: [991726](https://a.yandex-team.ru/review/991726)

**Comment by arcadia-devtools** [comment]
Check [991726](https://a.yandex-team.ru/review/991726) passed

**Pull request merged**